livedata/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -192,6 +205,18 @@ class ChatDomainImpl private constructor(
         initClean()
     }
 
+    private fun storeBackgroundSyncConfig(
+        backgroundSyncConfig: BackgroundSyncConfig,
+        context: Context
+    ) {
+        if (BackgroundSyncConfig.UNAVAILABLE != backgroundSyncConfig) {
+            EncryptedBackgroundSyncConfigStore(context.applicationContext).run {
```

Comment:
Could we inject this dependency instead of creating it here?
It will help us on test side

Reply:
I've extracted EncryptedBackgroundSyncConfigStore creation into SyncProvider class. 
If you have a better concept of implementing DI in this project, we can discuss and refactor later.